src/test/com/fulcrologic/statecharts/data_model/working_memory_data_model_spec.cljc

```diff
@@ -62,7 +62,18 @@
 
       (assertions
         "removes the value"
-        (sp/get-at DM (context :ROOT) :z) => nil))))
+        (sp/get-at DM (context :ROOT) :z) => nil))
+
+    (component "load-data"
+      (let [slurp-able-data (char-array (pr-str {::loaded :data-loaded-from-src}))
+            more-slurp-able-data (char-array (pr-str {::loaded :data-loaded-from-src-into-context}))]
+
+        (sp/load-data DM (context :ROOT) slurp-able-data)
+        (sp/load-data DM (context :A/a) more-slurp-able-data)
+
+        (assertions "Data loaded in"
```

Comment:
When writing tests, the general structure of english should compose into a sentence-like thing, when possible, for clarity about what you're testing.

"<Working memory model> <load-data> DOES WHAT?" In this case it ends up reading "Working memory model load data Data loaded in."

But that tells me, the person that has to debug things when the test breaks, nothing about what you expect. Does it merge it at the top level, or does it *replace* the entire data model? 

It also has the effect of making you *think* about what it *should* do, and perhaps this leads you to look at the spec. Maybe there are requirements, perhaps it can simply be invented by the implementor....in either case saying what you expect it to *do* is pretty important, and often leads to additional assertions, each of which often needs its own textual clause to clarify which of the expected behaviors you are proving.

As an additional note it is important for this language to say what it *does* and not the reverse.  Yeah, it doesn't paint my house red or launch a rocket...do I list all of the things it doesn't do? Proving negativity is very hard (you could screw up your measurement event, and see that "nothing" happened, but later find it was because you were looking for "nothing" in the wrong place).

Whereas, if you follow the rule of making assertions that actually expect a positive result (i.e. see a particular visible change) then you avoid that pitfall. So instead of "doesn't overwrite existing values" I would encourage "leaves existing, non-conflicting, values in place".  In this case the assertion is the same, but the language reminds you of the pitfall.

Reply:
OK, so when I re-read https://www.w3.org/TR/scxml/#datamodel and consider the implications:

1. The `datamodel` node can have *multiple* `data` children.
2. EACH child can have a `src` attribute

THEN it makes NO sense for loads to be a complete destruction of the data (in either of our data model implementations)

As an *extension* of the standard, we *could* allow an additional property on a `data` node to indicate a desired behavior (overwrite vs merge), but I think the default should be merge.